controller/status.go

```diff
@@ -277,6 +263,11 @@ var (
 	}
 )
 
+// Aggregator provides methods for aggregating status.
+type Aggregator struct {
```

Comment:
either make the functions on this type public as well or hide the type. A public type with no functions is useless.